core/utils.js

```diff
@@ -587,6 +587,47 @@ Blockly.utils.getBlockTypeCounts = function(block, opt_stripFollowing) {
   return typeCountsMap;
 };
 
+var resizeFromKeyboard = true;
+var initialWidth = 0;
+var debounceTimeout = null;
+
+/**
+ * Called in the resize event handler to sense if the source
+ * is from an on screen keyboard or user.
+ * @param {!Blockly.WorkspaceSvg} workspace Any workspace in the SVG.
+ */
+Blockly.utils.resizeTracker = function(workspace) {
+  if (debounceTimeout) {
+    clearTimeout(debounceTimeout);
+  }
+  debounceTimeout = setTimeout(function(){
+    resizeFromKeyboard = true;
+    initialWidth = workspace.width;
+  }, 1000);
+
+  if (resizeFromKeyboard) {
+    // If the on-screen keyboard is causing the resize, only the
+    // height will change.
+    if (workspace.width != initialWidth) {
```

Comment:
won't this trigger if the user resizes the browser window vertically (with a mouse) as well, though?

Reply:
ah yeah that's true. Hmmm maybe I could look at how fast it is? Since the keyboard probably changes it fast and in only one direction

Reply:
is this function https://github.com/microsoft/pxt-blockly/blob/develop/core/dropdowndiv.js#L730 not successfully preserving the visibility of the widget/dropdown divs?

Reply:
OO I didn't know about that function. It's not working I think because it's being hidden first (but I gotta check that just going off my gut)

Reply:
(oh yes i looked into this and even if we did reposition the dropdown correctly after, it doesn't matter since we've already refocused onto the workspace, and if we tried to focus back onto the input we'd start the on screen keyboard popping up/down infinitely)

and for the original comment, I think we're actually okay from testing it out. if the window gets too small, the dropdown gets closed, and if we're using touch, the onscreen keyboard will actually block the bottom edge so it can't be used to resize

Reply:
hmm maybe i'm missing something? if i resize the workspace up horizontally won't that permanently set "resizeFromKeyboard" to true?

Reply:
hmm ok but i guess, worst case is the widgets will remain onscreen and not be hidden, which is probably not terrible

Reply:
permanently for that resize, not permanently forever (not sure if that was what you mean?)

since the only time we're checking to see if it's resizing from the keyboard is when we're deciding to hide the widget/dropdown, the worst that could happen is it wouldn't properly hide it when it gets too small to fit -- but for some reason when I was testing it, the dropdown was still automatically closed when it wouldnt fit anymore

![vertical resize](https://user-images.githubusercontent.com/18712271/104667721-97134f00-568b-11eb-8926-fc4da463a3f5.gif)

not sure why it works though???

or I might not be understanding your concern correctly >>

Reply:
the one other case we have for vertical resizing is the more/less for the tutorial step, so if the tutorial expands the widget might cover the tutorial card, but that should be pretty easy to recover from also

Reply:
oooo i see i'll try it out though!

Reply:
hmmm yeah so it doesn't close with the tutorials, but I just tried it out on live too and that doesn't work either :\

Reply:
not a regression! seems fine to me then :D